File: roman_numerals/javascript/test/RomanNumeralsTest.js
test('Convert numeral to roman', function () {
  equal(romanNumerals(1), 'I');
  equal(romanNumerals(2), 'II');
  equal(romanNumerals(3), 'III');

  equal(romanNumerals(4), 'IV');
  equal(romanNumerals(5), 'V');
  equal(romanNumerals(8), 'VIII');

  equal(romanNumerals(9), 'IX');
  equal(romanNumerals(10), 'X');
  equal(romanNumerals(39), 'XXXIX');
  
  equal(romanNumerals(90), 'XC');
  equal(romanNumerals(100), 'C');
  equal(romanNumerals(399), 'CCCXCIX');

  equal(romanNumerals(399), 'CCCXCIX');
  equal(romanNumerals(399), 'CCCXCIX');
  equal(romanNumerals(399), 'CCCXCIX');

  equal(romanNumerals(400), 'CD');
  equal(romanNumerals(500), 'D');
  equal(romanNumerals(899), 'DCCCXCIX');

  equal(romanNumerals(900), 'CM');
  equal(romanNumerals(1000), 'M');
  equal(romanNumerals(3999), 'MMMCMXCIX');
});

test('Numbers must be in range 1..3999', function () {
  raises(function() {
      romanNumerals(-1);
  }, 'Numeral has no roman representation');

  raises(function() {
      romanNumerals(-0);
  }, 'Numeral has no roman representation');

  raises(function() {
      romanNumerals(4000);
  }, 'Numeral has no roman representation');
});

File: benchmark_test_frameworks/rspec/test/roman_numerals_test.rb
require 'roman_numerals'

describe RomanNumerals do
  it "something" do
   RomanNumerals.to_roman(1).should == 'I'
   RomanNumerals.to_roman(2).should == 'II'
   RomanNumerals.to_roman(3).should == 'III'
   RomanNumerals.to_roman(4).should == 'IV'
   RomanNumerals.to_roman(5).should == 'V'
   RomanNumerals.to_roman(8).should == 'VIII'

   RomanNumerals.to_roman(9).should == 'IX'
   RomanNumerals.to_roman(10).should == 'X'
   RomanNumerals.to_roman(39).should == 'XXXIX'

   RomanNumerals.to_roman(40).should == 'XL'
   RomanNumerals.to_roman(50).should == 'L'
   RomanNumerals.to_roman(89).should == 'LXXXIX'

   RomanNumerals.to_roman(90).should == 'XC'
   RomanNumerals.to_roman(100).should == 'C'
   RomanNumerals.to_roman(399).should == 'CCCXCIX'

   RomanNumerals.to_roman(400).should == 'CD'
   RomanNumerals.to_roman(500).should == 'D'
   RomanNumerals.to_roman(899).should == 'DCCCXCIX'

   RomanNumerals.to_roman(900).should == 'CM'
   RomanNumerals.to_roman(1000).should == 'M'
   RomanNumerals.to_roman(3999).should == 'MMMCMXCIX'
 end

  it "raises" do
    lambda { RomanNumerals.to_roman(-1) }.should raise_error
    lambda { RomanNumerals.to_roman(0) }.should raise_error
    lambda { RomanNumerals.to_roman(4000) }.should raise_error
  end
end
